internal/compiler-bridge/src/main/scala_2.11-12/xsbt/ConsoleInterface.scala

```diff
@@ -0,0 +1,105 @@
+/*
+ * Zinc - The incremental compiler for Scala.
+ * Copyright 2011 - 2017, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * This software is released under the terms written in LICENSE.
+ */
+
+package xsbt
+
+import xsbti.Logger
+import scala.tools.nsc.interpreter.{ ILoop, IMain, InteractiveReader, NamedParam }
+import scala.tools.nsc.reporters.Reporter
+import scala.tools.nsc.{ GenericRunnerCommand, Settings }
+
+class ConsoleInterface {
+  def commandArguments(
+      args: Array[String],
+      bootClasspathString: String,
+      classpathString: String,
+      log: Logger
+  ): Array[String] =
+    MakeSettings.sync(args, bootClasspathString, classpathString, log).recreateArgs.toArray[String]
+
+  def run(
+      args: Array[String],
+      bootClasspathString: String,
+      classpathString: String,
+      initialCommands: String,
+      cleanupCommands: String,
+      loader: ClassLoader,
+      bindNames: Array[String],
+      bindValues: Array[Any],
+      log: Logger
+  ): Unit = {
+    lazy val interpreterSettings = MakeSettings.sync(args.toList, log)
+    val compilerSettings = MakeSettings.sync(args, bootClasspathString, classpathString, log)
+
+    log.info(Message("Starting scala interpreter..."))
+    log.info(Message(""))
+
+    val loop = new ILoop {
+      override def createInterpreter() = {
+        if (loader ne null) {
+          in = InteractiveReader.apply()
+          intp = new IMain(settings) {
+            override protected def parentClassLoader =
+              if (loader eq null) super.parentClassLoader else loader
+
+            override protected def newCompiler(settings: Settings, reporter: Reporter) =
+              super.newCompiler(compilerSettings, reporter)
+          }
+          intp.setContextClassLoader()
+        } else
+          super.createInterpreter()
+
+        for ((id, value) <- bindNames zip bindValues)
+          intp.quietBind(NamedParam.clazz(id, value))
```

Comment:
Due to the regressions we had I'd rather we use

```scala
NamedParamClass(id, value.asInstanceOf[AnyRef].getClass.getName, value)
```

rather than `NamedParam.clazz`.

Reply:
Could you link to the regressions?
FWIW, this file was copy-pasted straight from `src/main/xsbt/ConsoleInterface.scala` because `ConsoleInterface.scala` is now split into three Scala specific pieces. For the sake of keeping this PR simple, any changes to 2.12 I think should go into another PR.

Reply:
My mistake. What you have is the right implementation.

I was reading 1.x ConsoleInterface: https://github.com/sbt/zinc/blob/1.x/internal/compiler-bridge/src/main/scala/xsbt/ConsoleInterface.scala#L57
which doesn't include my fixes, that are in 1.0.x ConsoleInterface: https://github.com/sbt/zinc/blob/1.0.x/internal/compiler-bridge/src/main/scala/xsbt/ConsoleInterface.scala#L57

We should really merge 1.0.x into 1.x to avoid this confusion.

Anyways these are the relevant commits:
* https://github.com/sbt/zinc/commit/0e4bb2cb05530b1680ee9899d562850d9b6f23c7
* https://github.com/sbt/zinc/commit/33d2e6897ccdae8bf9f513e7ae4e044743bfd691